tests/test_embed/test_interpreter.cpp

```diff
@@ -525,6 +526,15 @@ TEST_CASE("Per-Subinterpreter GIL") {
         }
         T_REQUIRE(caught);
 
+        // widget_module did provide the per_interpreter_gil tag, so it this does not throw
+        try {
+            py::module_::import("widget_module");
+            caught = false;
+        } catch (pybind11::error_already_set &) {
+            caught = true;
+        }
+        T_REQUIRE(!caught);
```

Comment:
This will mask the unexpected exception, which will make debugging harder.
Could this simply be
```
        py::module_::import("widget_module");
```
?

Reply:
Yes, it could.  If it fails is still kind of ugly ... since it's in a different thread it can't use normal `REQUIRE` or `REQUIRE_NOTHROW`.